src/EditorFeatures/Core/Implementation/IntelliSense/Completion/AsyncCompletionService.cs

```diff
@@ -57,31 +66,49 @@ public bool TryGetController(ITextView textView, ITextBuffer subjectBuffer, out
         {
             AssertIsForeground();
 
-            // check whether this feature is on.
-            if (!subjectBuffer.GetFeatureOnOffOption(InternalFeatureOnOffOptions.CompletionSet))
+            if (!UseLegacyCompletion(textView, subjectBuffer))
```

Comment:
is this purely for testing purposes... or is this something we would expect users would have control of?  do we intend to run A/B experiements here?  is there an expected timeline for removal?

Reply:
The current plan is as follows: 
1. merge this under a feature flag: the new completion will be disabled for all users.
2. one will be able to enable the new completion by turning on the feature flag.
3. fix all the issues and get signoff from all partner teams.
4. remove the feature flag

---
In reply to: [222071627](https://github.com/dotnet/roslyn/pull/29016#discussion_r222071627) [](ancestors = 222071627)